File: nandcpp/Count the Hidden Sequences.cpp
class Solution {
public:
    int numberOfArrays(vector<int>& d, int lower, int upper) 
    {
        long maxi, mini, sum;
        maxi = mini = sum = 0;
        
        for(auto i : d)
        {
            sum += i;
            
            maxi = max(maxi, sum);
            mini = min(mini, sum);
        }
        
        return max(0L, (upper-maxi)+(mini-lower)+1);
    }
};

File: nandincpp/Count the Hidden Sequences.java
class Solution {
    public int numberOfArrays(int[] d, int lower, int upper) 
    {
        long sum = 0, mini = 0, maxi = 0;
        
        for(int i : d)
        {
            sum += i;
            
            maxi = Math.max(maxi, sum);
            mini = Math.min(mini, sum);
        }
        
        return (int)Math.max(0, (upper-maxi)+(mini-lower)+1);
    }
}